client/post-editor/editor-featured-image/index.jsx

```diff
@@ -115,6 +127,7 @@ export default React.createClass( {
 				borderless
 				compact
 				className={ classes }>
+				{ site && featuredImageId && <QueryMedia siteId={ site.ID } mediaId={ featuredImageId } /> }
```

Comment:
Usage here looks a bit awkward. I'd expect to be able to pass through falsey values to a QueryComponent, since props might not be available on a clean cache. 

I'd recommend updating QueryMedia to take an explicit boolean prop that toggles a single/query request.

At the moment it toggles off the falsy value of mediaId:
const single = !! mediaId;
if ( single ) {
	this.props.requestMediaItem( siteId, mediaId );
} else {
	this.props.requestMedia( siteId, query );
}
```

In another PR I'd recommend updating this so we have usages like the following, and we stop guessing on intent.
```jsx
<QueryMedia siteId={ siteId } query={ query } />
<QueryMedia siteId={ siteId } mediaId={ mediaId } single />
```